resources/shaders/ambient_occlusion.hlsl

```diff
@@ -3,7 +3,7 @@
 
 RWTexture2D<float4> output : register(u0); // x: AO value
 Texture2D gbuffer_normal : register(t1);
-Texture2D gbuffer_depth : register(t2);
+Texture2D gbuffer_position: register(t2);
```

Comment:
What is the reason for this change? Also I want to see a performance comparison with deferred master and deferred your branch.

Reply:
with this change I mean using gbuffer_position instead of depth.

Reply:
Due to the non-linear nature of the depth buffer, it can't provide sufficient accuracy for RTAO. This was the primary reason for our insane self-intersection. I will test the performance of this change to be able to make a better-educated opinion.

Reply:
Using a linear depth buffer would be less bandwidth heavy, as the buffer you're using is an RGBA16 bit buffer, while a linear depth buffer would only require a single-channel buffer.
That said, I'm going to accept this change and write it down for the optimization phase, if time allows, as I would like you to be able to move to your next task.